app/code/Magento/Checkout/Test/Unit/Controller/Account/CreateTest.php

```diff
@@ -66,53 +83,48 @@ protected function setUp()
 
     public function testExecuteAddsSessionMessageIfCustomerIsLoggedIn()
     {
-        $jsonFactoryMock = $this->createMock(\Magento\Framework\Controller\Result\JsonFactory::class);
-        $this->objectManagerMock->expects($this->once())
-            ->method('get')
-            ->with(\Magento\Framework\Controller\Result\JsonFactory::class)
-            ->willReturn($jsonFactoryMock);
-        $jsonMock = $this->createMock(\Magento\Framework\Controller\Result\Json::class);
-        $jsonFactoryMock->expects($this->once())->method('create')->willReturn($jsonMock);
-
-        $this->customerSession->expects($this->once())->method('isLoggedIn')->will($this->returnValue(true));
-
-        $jsonMock->expects($this->once())
+        $resultJson = '{"errors": "true", "message": "Customer is already registered"}';
+        $this->customerSession->expects($this->once())
+            ->method('isLoggedIn')
+            ->will($this->returnValue(true));
+        $this->resultFactory->expects($this->once())
+            ->method('create')
+            ->with(ResultFactory::TYPE_JSON)
+            ->willReturn($this->resultPage);
+        $this->resultPage->expects($this->once())
             ->method('setData')
             ->with(
                 [
                     'errors' => true,
                     'message' => __('Customer is already registered')
                 ]
-            )->willReturnSelf();
-        $this->action->execute();
+            )->willReturn($resultJson);
+        $this->assertEquals($resultJson, $this->action->execute());
     }
 
     public function testExecute()
     {
-        $jsonFactoryMock = $this->createMock(\Magento\Framework\Controller\Result\JsonFactory::class);
-        $this->objectManagerMock->expects($this->once())
-            ->method('get')
-            ->with(\Magento\Framework\Controller\Result\JsonFactory::class)
-            ->willReturn($jsonFactoryMock);
-        $jsonMock = $this->createMock(\Magento\Framework\Controller\Result\Json::class);
-        $jsonFactoryMock->expects($this->once())->method('create')->willReturn($jsonMock);
-
         $this->customerSession->expects($this->once())->method('isLoggedIn')->will($this->returnValue(false));
         $this->checkoutSession->expects($this->once())->method('getLastOrderId')->will($this->returnValue(100));
         $customer = $this->createMock(\Magento\Customer\Api\Data\CustomerInterface::class);
-        $this->orderCustomerService->expects($this->once())->method('create')->with(100)->will(
-            $this->returnValue($customer)
-        );
-
-        $jsonMock->expects($this->once())
+        $this->orderCustomerService->expects($this->once())
+            ->method('create')
+            ->with(100)
+            ->will($this->returnValue($customer));
+
+        $resultJson = '{"errors": "false", "message": "A letter with further instructions will be sent to your email."}';
```

Comment:
@dmanners Please can you split this up so the line doesn't exceed 120 characters? I'd do it myself but my VM pretends to by dying atm so I cannot run my tests locally *sigh*. ;)

Reply:
sure I can sort that out today